### ISSUE TITLE
Free rotation of the 3D view via keyboard

### ISSUE THREAD

**Issue by DLFW** [openscad/openscad#3231]
The 3D view can be zoomed and brought to some basic positions via keyboard shortcuts. But it cannot be freely rotated.

**Request:** Add keyboard shortcuts for rotating the 3D view around all three axes and for each in both directions.

Optional: Provide shortcuts for small and big rotation steps, maybe 1° and 8°.

This request is a follow-up for #2977, which has been decided to be used as tracker for “configurable” shortcuts, only. (By @t-paul in https://github.com/openscad/openscad/pull/3226#issuecomment-589074106_.)

**Comment by t-paul** [comment]
The comment is on the PR and separate review of smaller things helps with limited time, but I guess it's even better to have a specific feature request too.

**Comment by thehans** [comment]
Something else that might be worth thinking about is the possibility of combining the shortcut key assignment functionality with input device configuration which already exists as "Buttons" tab in preferences.  
So that some of these new actions could be be assigned to joystick buttons, and shortcut keys could be set alongside buttons through some common interface.

**Comment by t-paul** [comment]
Indeed. If we go with the additional actions, it should be just a matter of registering those with the input driver manager.

**Comment by jordanbrown0** [comment]
Let's generalize this to rotate, pan, and zoom; they all seem to go together and it allows closing #4448 as a duplicate.

**Comment by jordanbrown0** [comment]
The "Button" mechanism doesn't currently support rotate/pan/zoom; it is largely tied to the actions available from the menus.  It does pick up a few actions from other places, notably the animation control panel and the "toggle perspective" action.  We could add actions for "rotate X by 1 degree", "rotate X by 15 degrees", "rotate Y by 1 degree", et cetera.  (Or maybe they could rotate by one of two or three preference-controlled amounts.)  Similarly for pan and zoom.  I don't think they would need to appear in any menu, other than the pull-down in the Button preference UI.

Note also that buttons are one-shot operations; they fire on press.  Continuing to hold the button has no effect.  For keyboards, autorepeat might help, depending on how it's done.

Or there could be a variation where buttons are treated as extremely-low-resolution axis controls that can report only 0.0 and +1.0; the normal axis-mapping stuff could then map that to pan, rotate, and zoom, with a configured gain.
